blocks/graph.py

```diff
@@ -528,3 +531,107 @@ def apply_dropout(computation_graph, variables, drop_prob, rng=None,
         replacement.tag.replacement_of = variable
 
     return computation_graph.replace(replacements)
+
+
+def apply_batch_normalization(computation_graph, variables, gammas,
```

Comment:
Specifying gammas and betas every time is somewhat annoying. I'd like them to be optional, set to a default value if an explicit value is not provided.

I guess the following behavior might not be optimal in some cases, but it should mostly work:
``` python
def apply_batch_normalization(computation_graph, variables, gammas=None, betas=None, axis=0,
                              epsilon=1e-7, use_population=False):
    if not betas:
        betas = []
        for i, v in enumerate(variables):
            betas.append(shared_floatx(numpy.zeros(v.shape[-1]), name='betas_{}'.format(i+1)))
    if not gammas:
        gammas = []
        for i, v in enumerate(variables):
            gammas.append(shared_floatx(numpy.ones(v.shape[-1]), name='gammas_{}'.format(i+1)))
```